objectherkenning_openbare_ruimte/data_delivery_pipeline/components/data_delivery.py

```diff
@@ -0,0 +1,152 @@
+import csv
+import os
+from typing import Dict, List
+
+from cvtoolkit.helpers.file_helpers import delete_file, find_image_paths
+
+from objectherkenning_openbare_ruimte.data_delivery_pipeline.components.iot_handler import (
+    IoTHandler,
+)
+from objectherkenning_openbare_ruimte.settings.settings import (
+    ObjectherkenningOpenbareRuimteSettings,
+)
+
+
+class DataDelivery:
+    def __init__(
+        self, images_folder: str, detections_folder: str, metadata_folder: str
+    ):
+        """
+
+        Parameters
+        ----------
+        images_folder
+            Folder containing the blurred images with containers detected
+        detections_folder
+            Folder containing txt files with detections per image
+        metadata_folder
+            Folder containing the metadata files in csv format
+        """
+        self.images_folder = images_folder
+        self.detections_folder = detections_folder
+        self.metadata_folder = metadata_folder
+        self.iot_settings = ObjectherkenningOpenbareRuimteSettings.get_settings()[
+            "azure_iot"
+        ]
+
+    def run_pipeline(self):
+        """ "
+        Runs the data delivery pipeline:
+            - matches metadata to images;
+            - delivers the data to Azure;
+            - deletes the delivered data.
+        """
+        print(f"Running data delivery pipeline on {self.images_folder}..")
+        images_and_frames = self._match_metadata_to_images()
+        self._deliver_data(images_and_frames=images_and_frames)
+        self._delete_data(images_and_frames=images_and_frames)
+
+    def _match_metadata_to_images(self) -> Dict[str, List[str]]:
+        """
+        Creates a csv file containing only the metadata of images with containers.
+        Returns video names and which frames contains containers.
+
+        Returns
+        -------
+        Dictionary containing as key a video name and as values the number of frames containing containers.
+        """
+        images_paths = find_image_paths(root_folder=self.images_folder)
+        videos_and_frames = self._get_videos_and_frame_numbers(
+            images_paths=images_paths
+        )
+        self._create_filtered_metadata_files(videos_and_frames=videos_and_frames)
+        return videos_and_frames
+
+    def _create_filtered_metadata_files(
+        self, videos_and_frames: Dict[str, List[str]]
+    ) -> None:
+        """
+        Creates a metadata file containing only metadata information of frames with containers.
+
+        Parameters
+        ----------
+        videos_and_frames
+            Dictionary containing as key a video name and as values the number of frames containing containers.
+        """
+        for video_name, frame_numbers in videos_and_frames.items():
+            filtered_rows = []
+            with open(f"{self.metadata_folder}/{video_name}.csv") as fd:
```

Comment:
maybe put this in a try block, if we have missing metadata files, so a FileNotFoundError, then another general Exception